### PR TITLE
feat: handle missing disbursements resulting in empty quota

### PR DESCRIPTION
**Pull request by dalsontws**
[Notion link](https://www.notion.so/FE-Error-handling-of-whitelisted-beneficiaries-that-do-not-have-any-disbursements-45e33974693e43e6aff9945fa10129b7) 

This PR handles the possibility where quota returned is empty for `v2/quota` endpoint, i.e. :
```
{
  "remainingQuota": Array [],
  "globalQuota": Array [],
  "localQuota": Array [],
}
```
which could happen when a beneficiary is whitelisted, but not disbursed any credits.

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [ ] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [x] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [x] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

File: src/hooks/useCart/useCart.test.tsx [hunk 1]
```diff
@@ -297,6 +306,58 @@ describe("useCart", () => {
...
+              features: {
+                ...defaultFeatures,
+                apiVersion: "v2",
+                campaignName: "Some Campaign Name",
+              },
+              policies: [defaultProducts[0]],
+              c13n: {},
+            }}
+          >
+            <ProductContextProvider products={defaultProducts}>
```
Comment (andrehadianto) [inline src/hooks/useCart/useCart.test.tsx:339]: Since this is the same as `wrapper`, should this be replaced with `{ wrapper }`?

**Review by andrehadianto — CHANGES_REQUESTED** [review]
1 comment!

File: src/hooks/useCart/useCart.test.tsx (hunk 1 above)
Comment (dalsontws) [reply src/hooks/useCart/useCart.test.tsx:339]: hmm i tried this but it doesnt seem to work, not sure if it's possible to perform abstraction here tho! any idea?

**Review by andrehadianto — APPROVED** [review]
LGTM 🚀

**Pull request merged**